src/main/java/org/microshed/intellij/wizard/MicroProfileSelectionStep.java

```diff
@@ -0,0 +1,251 @@
+        if (specMatrix == null) {
+            rootPanel.add(new JLabel("Fetching MicroProfile specs from " + MicroProfileModuleBuilder.STARTER_REST_BASE_URL + " failed."));
+            return rootPanel;
+        }
+
+        rootPanel = JBUI.Panels.simplePanel(0, 10)
+                .addToTop(createTopPanel())
+                .addToCenter(createCenterPanel())
+                .addToBottom(createBottomPanel());
+
+        return rootPanel;
+    }
+
+    private JPanel createTopPanel() {
+        JPanel topPanel = new JPanel();
+        GridBagLayout topPanelLayout = new GridBagLayout();
+        topPanel.setLayout(topPanelLayout);
+
+        GridBagConstraints topPanelLayoutConstraint = new GridBagConstraints();
+        topPanelLayoutConstraint.fill = GridBagConstraints.HORIZONTAL;
+        topPanelLayoutConstraint.insets = JBUI.insets(2);
+        topPanelLayoutConstraint.weightx = 0.25;
+        topPanelLayoutConstraint.gridx = 0;
+
+        groupIdTextField = new JTextField("com.example");
+        topPanelLayoutConstraint.gridy = 0;
+        topPanel.add(new LabeledComponent("GroupId", groupIdTextField), topPanelLayoutConstraint);
+
+        artifactIdTextField = new JTextField("demo");
+        topPanelLayoutConstraint.gridy++;
+        topPanel.add(new LabeledComponent("ArtifactId", artifactIdTextField), topPanelLayoutConstraint);
+
+
+        mpVersionsComboBox = new ComboBox<String>(specMatrix.getConfigs().keySet().toArray(new String[0]));
+        mpVersionsComboBox.setRenderer(new MPVersionComboBoxRenderer());
+        mpVersionsComboBox.addItemListener(event -> {
+            if (event.getStateChange() == ItemEvent.SELECTED) {
+                String selectedVersion = (String) event.getItem();
+
+                //  Update servers list and specs tree
+                List<String> supportedServers = specMatrix.getConfigs().get(selectedVersion).getSupportedServers();
+                mpServersComboBox.removeAllItems();
+                supportedServers.forEach(mpServersComboBox::addItem);
+
+                List<String> specs = specMatrix.getConfigs().get(selectedVersion).getSpecs();
+                mpSpecsCheckboxTree.setModel(specMatrix.getParsedDescription().stream()
+                        .filter(spec -> specs.contains(spec.getName()))
+                        .collect(Collectors.toList()));
+            }
+        });
+        topPanelLayoutConstraint.gridy++;
+        topPanel.add(new LabeledComponent("MicroProfile Versions", mpVersionsComboBox), topPanelLayoutConstraint);
+
+
+        mpServersComboBox = new ComboBox<String>(specMatrix.getConfigs().get(mpVersionsComboBox.getItemAt(mpVersionsComboBox.getSelectedIndex()))
+                .getSupportedServers().toArray(new String[0]));
+        mpServersComboBox.setRenderer(new MPServerComboBoxRenderer());
+
+        topPanelLayoutConstraint.gridy++;
+        topPanel.add(new LabeledComponent("MicroProfile Servers", mpServersComboBox), topPanelLayoutConstraint);
```

Comment:
Should say "MicroProfile Implemenations"

Reply:
Fixed.